src/OpenFeature/Hooks/LoggingHook.cs

```diff
@@ -0,0 +1,182 @@
+        }
+
+        /// <inheritdoc/>
+        public override ValueTask AfterAsync<T>(HookContext<T> context, FlagEvaluationDetails<T> details, IReadOnlyDictionary<string, object>? hints = null, CancellationToken cancellationToken = default)
+        {
+            var evaluationContext = this._includeContext ? context.EvaluationContext : null;
+
+            var content = new LoggingHookContent(
+                context.ClientMetadata.Name,
+                context.ProviderMetadata.Name,
+                context.FlagKey,
+                context.DefaultValue?.ToString(),
+                evaluationContext);
+
+            this.HookAfterStageExecuted(content);
+
+            return base.AfterAsync(context, details, hints, cancellationToken);
+        }
+
+        [LoggerMessage(
+            Level = LogLevel.Debug,
+            Message = "Before Flag Evaluation {Content}")]
+        partial void HookBeforeStageExecuted(LoggingHookContent content);
+
+        [LoggerMessage(
+            Level = LogLevel.Error,
+            Message = "Error during Flag Evaluation {Content}")]
+        partial void HookErrorStageExecuted(LoggingHookContent content);
+
+        [LoggerMessage(
+            Level = LogLevel.Debug,
+            Message = "After Flag Evaluation {Content}")]
+        partial void HookAfterStageExecuted(LoggingHookContent content);
+
+        /// <summary>
+        /// Generates a log string with contents provided by the <see cref="LoggingHook"/>.
+        /// <para>
+        ///     Specification for log contents found at https://github.com/open-feature/spec/blob/d261f68331b94fd8ed10bc72bc0485cfc72a51a8/specification/appendix-a-included-utilities.md#logging-hook
+        /// </para>
+        /// </summary>
+        internal class LoggingHookContent
+        {
+            private readonly string _domain;
+            private readonly string _providerName;
+            private readonly string _flagKey;
+            private readonly string _defaultValue;
+            private readonly EvaluationContext? _evaluationContext;
+
+            public LoggingHookContent(string? domain, string? providerName, string flagKey, string? defaultValue, EvaluationContext? evaluationContext = null)
+            {
+                this._domain = string.IsNullOrEmpty(domain) ? "missing" : domain!;
+                this._providerName = string.IsNullOrEmpty(providerName) ? "missing" : providerName!;
+                this._flagKey = flagKey;
+                this._defaultValue = string.IsNullOrEmpty(defaultValue) ? "missing" : defaultValue!;
+                this._evaluationContext = evaluationContext;
+            }
+
+            public override string ToString()
+            {
+                var stringBuilder = new StringBuilder();
+
+                stringBuilder.Append("Domain:");
+                stringBuilder.Append(this._domain);
+                stringBuilder.Append(Environment.NewLine);
```

Comment:
This code can be simplified. The same applies for the next lines.
```suggestion
                stringBuilder.AppendLine($"Domain: {this._domain}");
```

Reply:
I forgot about this method on StringBuilder - I've switched to `stringBuilder.AppendLine` instead of manually Appending `Environment.NewLine`